src/nwb/NWBFile.cpp

```diff
@@ -142,14 +146,88 @@ Status NWBFile::createElectricalSeries(
     electricalSeries->initialize();
     recordingContainers->addContainer(std::move(electricalSeries));
     containerIndexes.push_back(recordingContainers->containers.size() - 1);
+  }
 
-    // Add electrode information to electrode table (does not write to datasets
-    // yet)
-    elecTable.addElectrodes(channelVector);
+  if (!electrodeTableCreated) {
+    // Add electrode information to table (does not write to datasets yet)
+    for (const auto& channelVector : recordingArrays) {
+      elecTable->addElectrodes(channelVector);
+    }
+
+    // write electrode information to datasets
+    elecTable->finalize();
   }
 
-  // write electrode information to datasets
-  elecTable.finalize();
+  return Status::Success;
+}
+
+Status NWBFile::createSpikeEventSeries(
+    std::vector<Types::ChannelVector> recordingArrays,
+    const SizeType numSamples,
+    const BaseDataType& dataType,
+    RecordingContainers* recordingContainers,
+    std::vector<SizeType>& containerIndexes)
+{
+  if (!io->canModifyObjects()) {
+    return Status::Failure;
+  }
+
+  // Setup electrode table if it was not yet created
+  bool electrodeTableCreated =
+      io->objectExists(ElectrodeTable::electrodeTablePath);
+  if (!electrodeTableCreated) {
+    elecTable = std::make_unique<ElectrodeTable>(io);
+    elecTable->initialize();
+  }
+
+  // Create continuous datasets
+  for (const auto& channelVector : recordingArrays) {
+    // Setup electrodes and devices
+    std::string groupName = channelVector[0].groupName;
+    std::string devicePath = "/general/devices/" + groupName;
+    std::string electrodePath = "/general/extracellular_ephys/" + groupName;
+    std::string spikeEventSeriesPath = acquisitionPath + groupName;
+
+    Device device = Device(devicePath, io, "description", "unknown");
+    device.initialize();
+
+    ElectrodeGroup elecGroup =
+        ElectrodeGroup(electrodePath, io, "description", "unknown", device);
+    elecGroup.initialize();
+
+    // Setup Spike Event Series datasets
+    SizeArray dsetSize;
+    SizeArray chunkSize;
+    if (channelVector.size() == 1) {
+      dsetSize = SizeArray {0, 0};
+      chunkSize = SizeArray {SPIKE_CHUNK_XSIZE, 1};
+    } else {
+      dsetSize = SizeArray {0, channelVector.size(), numSamples};
+      chunkSize = SizeArray {SPIKE_CHUNK_XSIZE, 1, 1};
+    }
```

Comment:
I think we here also need to check that the ``ElectrodeGroup`` and ``Device`` do not already exist. When recording a `SpikeEventSeries`, chances are we are also recording an `ElectricalSeries` with the raw data from the same device and channel at the same time. One option may be to have a variant of this function that would take a ``std::vector<ElectrodeGroup>`` instead of ``std::vector<Types::ChannelVector> recordingArrays`` as input. The same is true for ``createElectricalSeries``, depending on the order in which the user creates different TimeSeries for the same group of channels.

Reply:
I added a check whether the device exists for a specific channel group since in our current workflow the `Device` and `ElectrodeGroup` for each channel group always created at the same time. 

More long-term, I'm wondering if we might want to incorporate this to be a part of the `Container` constructor: check if the group / dataset exists already, create it if it doesn't exist, otherwise read it in? I'm not sure if that leads to more complications in other ways that I'm not thinking of....

Reply:
> More long-term, I'm wondering if we might want to incorporate this to be a part of the `Container` constructor: check if the group / dataset exists already, create it if it doesn't exist, otherwise read it in? I'm not sure if that leads to more complications in other ways that I'm not thinking of....

Seems reasonable, but I believe the check for whether the object exists or not should appear in the ``initialize`` method instead of the constructor. The constructor is responsible for creating the instance, but should not do any real work on write (nor read). The main difference between read and write is not how the Container object is being constructed, but what functions are called afterwards, i.e., on write we call ``initialize`` and on read the user will call methods to get lazy read access to properties of the ``Container``.